### ISSUE TITLE
Errors while running dev_setup.sh

### ISSUE THREAD

**Issue by Lynnesbian** [MycroftAI/mycroft-core#2339]
I encountered a few errors with `dev_setup.sh` on my Manjaro 5.3 box. I'm using the `master` branch.

Firstly, I received the error `Failed to set up virtualenv for mycroft, exiting setup.`. I was able to fix this by running `chmod +x .venv/bin/activate`.

I then an error from `/home/[my user]/.local/bin/pip` on the line `from pip._internal import main`, saying that the `pip` module couldn't be found. I fixed that by editing [line 344 of `dev_setup.sh`](https://github.com/MycroftAI/mycroft-core/blob/a77216714d41108e4a794f55db1719732a869d30/dev_setup.sh#L344) and removing the `--without-pip` flag.

`./start-mycroft.sh debug` ran successfully and my device read aloud an instruction to register it with `mycroft.ai`. I noticed many errors about being unable to find a package manager, but otherwise the device seemed to work fine.

**Comment by Lynnesbian** [comment]
While the pairing seemed to work fine, the device is unable to recognise any commands, responding to everything with variations of "I didn't catch that". I've noticed a lot of errors in `skills.log`, which I've uploaded [here](https://gist.github.com/Lynnesbian/38c69711407ee161d237e9deb48b63c8).

**Comment by Lynnesbian** [comment]
It seems as though the initial issue with `pip` being missing was caused by the line:
`curl https://bootstrap.pypa.io/3.3/get-pip.py | "${VIRTUALENV_ROOT}/bin/python" - 'pip==18.0.0'`
failing to install `pip` to `.venv/bin`. After the process exits, there is no `pip` executable created.

Removing the `--without-pip` flag ensures that a `pip` executable created in `.venv/bin` (along with `easy_install`), but this doesn't seem to work with Mycroft, given that it still complains with `Failed to launch package manager: Package manager not found!`

**Comment by forslund** [comment]
Hi, that last sounds like it comes from the msm skill installer and may be that there's no known package tool for manjaro...but most skills should install anyway...

Do you get any information about how /why the curl line fails?

**Comment by Lynnesbian** [comment]
It seems as though the issue with the missing package manager is caused by [this script](https://github.com/MycroftAI/pako/blob/master/pako/package_manager_data.py#L26), which only checks for `apt-get` and `eopkg` (whereas Manjaro uses `pacman`).

The curl script seems to run fine, but after it runs in `dev_setup.sh`, the script says that virtualenv setup failed.
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1603k  100 1603k    0     0   924k      0  0:00:01  0:00:01 --:--:--  924k
Collecting pip==18.0.0
  Using cached https://files.pythonhosted.org/packages/5f/25/e52d3f31441505a5f3af41213346e5b6c221c9e086a166f3703d2ddaf940/pip-18.0-py2.py3-none-any.whl
Collecting setuptools
  Using cached https://files.pythonhosted.org/packages/b2/86/095d2f7829badc207c893dd4ac767e871f6cd547145df797ea26baea4e2e/setuptools-41.2.0-py2.py3-none-any.whl
Collecting wheel<0.30
  Using cached https://files.pythonhosted.org/packages/8a/e9/8468cd68b582b06ef554be0b96b59f59779627131aad48f8a5bce4b13450/wheel-0.29.0-py2.py3-none-any.whl
Installing collected packages: pip, setuptools, wheel
Successfully installed pip-18.0 setuptools-41.2.0 wheel-0.29.0
Failed to set up virtualenv for mycroft, exiting setup.
$ ls .venv/bin
activate  activate.csh  activate.fish  python  python3
```

So while the curl script doesn't give any errors, it doesn't seem to install `pip` to the virtual environment.

```
$ curl https://bootstrap.pypa.io/3.3/get-pip.py | "${VIRTUALENV_ROOT}/bin/python" - 'pip==18.0.0'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1603k  100 1603k    0     0  1262k      0  0:00:01  0:00:01 --:--:-- 1262k
Collecting pip==18.0.0
  Using cached https://files.pythonhosted.org/packages/5f/25/e52d3f31441505a5f3af41213346e5b6c221c9e086a166f3703d2ddaf940/pip-18.0-py2.py3-none-any.whl
Installing collected packages: pip
  Found existing installation: pip 18.0
    Uninstalling pip-18.0:
      Successfully uninstalled pip-18.0
Successfully installed pip-18.0
You are using pip version 18.0, however version 19.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
$ pip --version
pip 18.0 from /home/lyra/.local/lib/python3.7/site-packages/pip (python 3.7)
```

Instead, the script installs `pip` to my `~/.local` folder, which overrides the system install of `pip` because it appears earlier in `$PATH`.

Running `pip` from the virtual environment worked fine, so I tried to start mycroft with `./start-mycroft.sh debug`. This time, it didn't even start, and gave me the error

```
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 183, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/usr/lib/python3.7/runpy.py", line 109, in _get_module_details
    __import__(pkg_name)
  File "/home/lyra/Downloads/src/mycroft-core/mycroft/__init__.py", line 17, in <module>
    from mycroft.api import Api
  File "/home/lyra/Downloads/src/mycroft-core/mycroft/api/__init__.py", line 19, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
```

So it seems as though changing `dev_setup.sh` to remove `--without-pip` allows it to get further and actually start the service, but all of the skills fail to load. Including the `--without-pip` flag means that Python looks for packages in the wrong location and fails to start Mycroft at all.

**Comment by forslund** [comment]
So that implies that none of the requirements are installed in the virtual environment...Sounds a bit like the venv doesn't get activated...

Can you try the following

```
source .venv/bin/activate
curl https://bootstrap.pypa.io/3.3/get-pip.py | ./.venv/bin/python" - 'pip==18.0.0'
```
Check that pip is installed in the correct folder (.venv/bin/)

if that's ok, run
```
pip install -r requirements.txt
pip install -r test-requirements.txt
msm default
```

**Comment by Lynnesbian** [comment]
Running the curl command works fine, but it doesn't install anything to `.venv/bin`.
```
$ source .venv/bin/activate
$ curl https://bootstrap.pypa.io/3.3/get-pip.py | "./.venv/bin/python" - 'pip==18.0.0'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1603k  100 1603k    0     0   244k      0  0:00:06  0:00:06 --:--:--  379k
Collecting pip==18.0.0
  Using cached https://files.pythonhosted.org/packages/5f/25/e52d3f31441505a5f3af41213346e5b6c221c9e086a166f3703d2ddaf940/pip-18.0-py2.py3-none-any.whl
Collecting setuptools
  Using cached https://files.pythonhosted.org/packages/b2/86/095d2f7829badc207c893dd4ac767e871f6cd547145df797ea26baea4e2e/setuptools-41.2.0-py2.py3-none-any.whl
Collecting wheel<0.30
  Using cached https://files.pythonhosted.org/packages/8a/e9/8468cd68b582b06ef554be0b96b59f59779627131aad48f8a5bce4b13450/wheel-0.29.0-py2.py3-none-any.whl
Installing collected packages: pip, setuptools, wheel
Successfully installed pip-18.0 setuptools-41.2.0 wheel-0.29.0
$ ls .venv/bin
activate  activate.csh  activate.fish  python  python3
```

**Comment by Lynnesbian** [comment]
I recently tried to reinstall Mycroft core again, using the latest git release.
I ran into a different issue with `dev_setup.sh`, namely a permission error (error 13) on the file `.venv/bin/__pycache__/wsdump.cpython-37.pyc`. The file belonged to `root`, which I fixed by executing `sudo chown [user]:[user] .`.

After that minor fix, `dev_setup.sh` completed successfully! I deleted the old `skills` directory and ran `./start_mycroft.sh debug`, and all skills installed and loaded successfully, except one that failed with the exception "Skill has been disabled by Mycroft", which I assume is intentional.

I still got all the msm errors about not finding a package manager, and I also get a few errors about messagebus failing to connect (it retries a few times, and then successfully connects - this might just be because the messagebus takes a while to start up on my Intel NUC) but everything seems to work fine apart from that.

If the msm package manager errors are caused by the script not knowing what Manjaro's package manager is, it's `yay`, which is a wrapper for `pacman` that implements some additional functionality. A potential fix would be changing a line that says something like 
```python
if distro == "Arch":
```
to
```python
if distro in ["Arch", "Manjaro"]:
```
allowing Manjaro to use `pacman`.

Apart from these minor issues, everything seems to be working properly, so I'll close this issue.
